dev/io.openliberty.webcontainer.servlet.6.1.internal_fat/test-applications/ReadWriteByteBuffer.war/src/readwritebytebuffer/servlets/ReadWriteByteBuffer.java

```diff
@@ -0,0 +1,153 @@
+        LOGENTER("doGET");
+
+        request = req;
+        response = resp;
+        responseSB = new StringBuilder();
+        sos = response.getOutputStream();
+
+        switch (request.getHeader("runTest")) {
+            case "testReadWriteByteBuffer" : testReadWriteByteBuffer(); break;
+            case "testReadNullByteBuffer" : testRedWriteNullByteBuffer(true); break;
+            case "testWriteNullByteBuffer" : testRedWriteNullByteBuffer(false); break;
+        }
+
+        if (!responseSB.isEmpty()) {
+            LOG(responseSB.toString());
+            sos.println(responseSB.toString());
+        }
+
+        LOGEXIT("doGET");
+    }
+
+
+    /*
+     * Test ServletInputStream.read(ByteBuffer) to read POST data.
+     * Then ServletOutputStream.write(ByteBuffer) to write data back to client for verification
+     */
+    private void testReadWriteByteBuffer() throws IOException{
+        String method = new Object(){}.getClass().getEnclosingMethod().getName();
+        LOGENTER(method);
+
+        int readIn;
+        try {
+            buffer = ByteBuffer.allocate(1024);
+
+            LOG("Read data using ServletInputStream.read(ByteBuffer)");
+            readIn = request.getInputStream().read(buffer);
+            LOG("read(ByteBuffer): read [" + readIn +"] bytes. DONE");
+
+            LOG("Write [" + buffer.remaining() + "] bytes to client using ServletOutputStream.write(ByteBuffer)");
+            sos.write(buffer);
+            LOG("write(ByteBuffer) DONE");
+        }
+        catch (Exception e) {
+            LOG("Exception ["+ e.getMessage() +"]");
+        }
+
+        LOGEXIT(method);
+    }
+
+    /*
+     * Test ServletInputStream.read(ByteBuffer) and ServletOutputStream.write(ByteBuffer) with NULL ByteBuffer to cause NPE
+     * test will catch, exam the message and report PASS/FAIL back to client for assert
+     *
+     * @param boolean read      - true - for read a null ByteBuffer
+     *                          - false - for write a null ByteBuffer
+     */
+    private void testRedWriteNullByteBuffer(boolean read) throws IOException{
+        String method = new Object(){}.getClass().getEnclosingMethod().getName();
+        LOGENTER(method);
+
+        try {
+            buffer = null;
+
+            if (read) {
+                LOG("ServletInputStream.read(ByteBuffer) with NULL ByteBuffer");
+                request.getInputStream().read(buffer);
+                LOG("read(ByteBuffer) DONE");
+            }
+            else {
+                LOG("ServletOutputStream.write(ByteBuffer) with NULL ByteBuffer");
+                sos.write(buffer);
+                LOG("write(ByteBuffer) DONE");
+            }
+        }
+        catch (Exception e) {
+            LOG("Exception ["+ (message = e.getMessage()) +"]");
+
+            if (message.contains("CWWWC0009E")) {
+                responseSB.append(method + " found the message with CWWWC0009E. PASS");
```

Comment:
Does this test pass right now?

This isn't a new message added by this PR.

> query.or.post.is.null=CWWWC0009E: The HTTP request query string or HTTP POST request data is null.
> query.or.post.is.null.explanation=The application tried to parse a null HTTP request query string or a null HTTP POST request body.
> query.or.post.is.null.useraction=The client must send a non-null HTTP request query string or a non-null HTTP POST request body.
> 
> read.write.bytebuffer.null=CWWWC0010E: An attempt to read from an InputStream operation or write to an OutputStream operation failed because a variable argument ByteBuffer instance was null.
> read.write.bytebuffer.null.explanation=The variable argument ByteBuffer instance that is passed to the read or write API must not be null.
> read.write.bytebuffer.null.useraction=Modify the application code to create a non-null ByteBuffer instance to pass to the InputStream or OutputStream operation.
> 
> read.write.failed.isReady.false=CWWWC0011E: An attempt to read from an InputStream operation or write to an OutputStream operation failed because the isReady API returns false.
> read.write.failed.isReady.false.explanation=If the isReady API returns false, you cannot call a read or write method.
> read.write.failed.isReady.false.useraction=Modify the application logic to make sure that the stream is ready for an InputStream or OutputStream operation.

`CWWWC0010E` and `CWWWC0010E` are for this PR, are we testing these new messages?